### PR TITLE
Publish April tag detection corners and tag center

### PR DESCRIPTION
**Pull request by 5yler**

### CODE REVIEW

File: apriltag_ros/msg/Point2D.msg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# This contains the position of a point in 2D pixel coordinates
+float64 u
+float64 v
```
Comment (duanyutong) [inline apriltag_ros/msg/Point2D.msg:3]: shouldn't these be `uint16_t` or `size_t`?

**Review by duanyutong — COMMENTED** [review]
`Pixel2D` seems a better name than `Point2D`. it can also carry the value(s) of the pixel

File: apriltag_ros/msg/AprilTagDetection.msg [hunk 2]
```diff
@@ -11,4 +11,11 @@ float64[] size
...
 # four corners of all member tags.
-geometry_msgs/PoseWithCovarianceStamped pose
\ No newline at end of file
+geometry_msgs/PoseWithCovarianceStamped pose
+
+# Tag center in pixel coordinates
+apriltag_ros/Point2D tag_center
+
+# Detected tag corners in pixel coordinates. Starting from bottom left going 
+# around anti-clockwise
```
Comment (wualbert) [inline apriltag_ros/msg/AprilTagDetection.msg:20]: This bottom left is the _tag's_ bottom left, not the bottom left in the camera frame, right?

File: apriltag_ros/src/common_functions.cpp [hunk 3]
```diff
@@ -340,6 +340,17 @@ AprilTagDetectionArray TagDetector::detectTags (
...
     tag_detection.size.push_back(tag_size);
+
+    // tag center in pixel coordinates
+    tag_detection.tag_center.u = detection->c[0];
+    tag_detection.tag_center.v = detection->c[1];
+
+    // tag corners in pixel coordinates
+    for (size_t idx{}; idx < 4; idx++) {
+      tag_detection.tag_corners[idx].u = detection->p[idx][0];
+      tag_detection.tag_corners[idx].v = detection->p[idx][1];
```
Comment (wualbert) [inline apriltag_ros/src/common_functions.cpp:351]: Are the tags sorted in this order by default?
> Detected tag corners in pixel coordinates. Starting from bottom left going 
> around anti-clockwise

**Review by wualbert — APPROVED** [review]
Minor comments, LGTM otherwise

File: apriltag_ros/msg/Point2D.msg (hunk 1 above)
Comment (5yler) [reply apriltag_ros/msg/Point2D.msg:3]: The library output is in floats not integer values. Example: 
```yaml
    tag_center: 
      u: 1167.1968160351246
      v: 435.0856950109363
    tag_corners: 
      - 
        u: 1196.1888427734375
        v: 399.94015502929693
      - 
        u: 1132.5440673828127
        v: 406.6687927246094
      - 
        u: 1139.0986328125002
        v: 469.14767456054693
      - 
        u: 1201.446044921875
        v: 463.17169189453136
```

File: apriltag_ros/src/common_functions.cpp (hunk 3 above)
Comment (5yler) [reply apriltag_ros/src/common_functions.cpp:351]: This is on a per-tag basis for each detection

File: apriltag_ros/msg/AprilTagDetection.msg (hunk 2 above)
Comment (5yler) [reply apriltag_ros/msg/AprilTagDetection.msg:20]: Yes. Consider 
image where the tags are upside down 
image

**Pull request merged**